src/components/page-search/page-search.js

```diff
@@ -1,13 +1,31 @@
 import React from 'react';
 import SubredditForm from './subreddit-form';
+import useFetchPosts from '../../hooks/use-fetch-posts';
 import * as S from './page-search.style';
 import Container from '../common/container';
 
-const SearchPage = () => (
-  <Container>
-    <S.Headline>Find the best time for a subreddit</S.Headline>
-    <SubredditForm />
-  </Container>
-);
+const SearchPage = () => {
+  const { status } = useFetchPosts();
+
+  return (
+    <Container>
+      <S.Headline>Find the best time for a subreddit</S.Headline>
+      <SubredditForm />
+
+      {
+       {
+         loading: <S.Spinner />,
+         error: (
+           <S.Error>
+             We could not get the posts from Reddit.
+             {' '}
+             <a rel="noopener noreferrer" href="https://www.redditstatus.com">Is Reddit down?</a>
+           </S.Error>
+         ),
+       }[status]
```

Comment:
Interesting approach to conditional rendering. Did you see that somewhere else? It makes sense but took me a moment to understand just because I've never seen it. So I'm not sure if this would be considered readable 😄

Reply:
Yeah, I picked up the idea along the way. I liked it for its readability, actually. Alternatively:

I often use the following approach.
```suggestion
      { status === 'loading' && <S.Spinner /> }
      { status === 'error' && <S.Error> ... </S.Error> }
      // status === 'resolved', etc.
```

Considering we care about three different values of `status`, it's a pretty good argument for a `switch` statement.
```suggestion
      {
        switch (status) {
          case 'resolved':
            return <SomeHeatmapComponent />
          case 'error':
            return <S.Error />
          default:
            return <S.Spinner />
        }
      }
```
It gets the job done but it's a bit more breathy than the object literal approach.

What do you recommend?

Reply:
I tend to go with the first option. But I don't think it really matters and is more of a personal preference. Your original way is interesting though and I only mentioned readability because it's not commonly used. You might write a blog post about it 😄